modules/core/src/main/java/org/apache/ignite/internal/processors/job/GridJobWorkerInterruptTimeoutObject.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2022 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.job;
+
+import org.apache.ignite.internal.processors.timeout.GridTimeoutObject;
+import org.apache.ignite.internal.util.typedef.internal.S;
+import org.apache.ignite.lang.IgniteUuid;
+
+/**
+ * Timeout object for delayed {@link GridJobWorker worker} interruption.
+ *
+ * <p>After calling {@link GridJobWorker#cancel} the worker should try to complete gracefully,
+ * if it doesn't then it will {@link Thread#interrupt interrupt} after some time.
+ */
+public class GridJobWorkerInterruptTimeoutObject implements GridTimeoutObject {
```

Comment:
`JobWorkerInterruptionTimeoutObject` - lets stick to this name. I don't see a lot of value in `Grid` prefix, and `Interruption` sounds more intuitive to me.